client/app/services/usage-graphs/usage-graphs.service.js

```diff
@@ -0,0 +1,79 @@
+/** @ngInject */
+export function UsageGraphsFactory() {
+  var service = {
+    cpuChartConfig: getCPUChartConfig,
+    memoryChartConfig: getMemoryChartConfig,
+    storageChartConfig: getStorageChartConfig,
+    convertBytestoGb: convertBytestoGb,
+  };
+
+  function getCPUChartConfig(used, total) {
+    let usedValue = 0;
+    let totalValue = 0;
+
+    if (angular.isDefined(used)) {
+      usedValue = used;
+      totalValue = total;
+    }
+
+    return {
```

Comment:
ditto the above... give it a few params, have it return an object, you'll be sitting pretty a mighty powerful itty bitty service

> his exact logic is repeated three times, recommend abstracting it out into a private (to this service) function